model/credential.py

```diff
@@ -54,13 +54,15 @@ class Credential(Base):
 
     @classmethod
     def lookup(self, _db, data_source, type, patron, refresher_method,
-               allow_persistent_token=False):
+               allow_persistent_token=False, allow_empty_token=False):
```

Comment:
This is an unrelated change stemming from my work on the RBdigital API. At one point due to a bug, my database ended up with a persistent Credential object where .credential was None. I think in general, if we see that this happens, the refresher method should be called -- the same as if the token had expired. Otherwise there's no way out of the problem.

However, theoretically `None` could be a valid value for a credential, so I added an argument that lets the user say so.